### ISSUE TITLE
Rule "no-unsafe-any" with Angular Injectable decorator false-positive error

### ISSUE THREAD

**Issue by felpel** [palantir/tslint#3784]
### Bug Report

- __TSLint version__: 5.9.1
- __TypeScript version__: 2.4.2
- __Running TSLint via__: Angular-CLI (ng lint)

#### TypeScript code being linted

```ts
import { Injectable } from "@angular/core";

@Injectable()
export class UserManagerFactory {
   // Factory logic omitted 
}
```

with `tslint.json` configuration:

```js
{
   "extends": [
       "tslint:recommended"
   ],
   "rules": {
      "no-unsafe-any": true
   }
}
```

#### Actual behavior

I get an error regarding the usage of `Injectable` as: `ERROR: 7:2      no-unsafe-any           Unsafe use of expression of type 'any'.`

#### Expected behavior

I'm not sure exactly how I should deal with this since we use intensively `Injectable` decorators in our Angular project. I don't want to disable this rule since it's very helpful to track bugs, but I might consider it if the proposed workaround is to disable this rule in every injected service class definition.

**Comment by wwarby** [comment]
I'm seeing the same behaviour in the same scenario, only difference is I'm using TypeScript 2.7.2.

**Comment by suchanlee** [comment]
what's the type of `Injectable` when imported?

**Comment by wwarby** [comment]
It's an empty interface:

`export interface Injectable {}`

The decorator is defined like this:

`export declare const Injectable: InjectableDecorator;`

Hovering over the decorator in vscode shows:

```
(alias) Injectable(): any
import Injectable
```

Hope that helps

**Comment by wwarby** [comment]
Hovering over the import statement in vscode shows:

```
(alias) interface Injectable
(alias) const Injectable: InjectableDecorator
import Injectable
```

**Comment by felpel** [comment]
@suchanlee I think @wwarby gave you the information that you needed for your initial investigation.

**Comment by JoshuaKGoldberg** [comment]
Root issue here is that `@angular/core/src/di/injectable.d.ts` has an `any` type as its return value when called as a function:

```typescript
 *
 * @publicApi
 */
export interface InjectableDecorator {
    /**
     * ...
     *
     */
    (): any;
    (options?: {
        providedIn: Type<any> | 'root' | null;
    } & InjectableProvider): any;
```

https://github.com/angular/angular/blob/4c2ce4e8ba4c5ac5ce8754d67bc6603eaad4564a/packages/core/src/di/injectable.ts#L48

Unfortunately, there's not much TSLint can do to alleviate this until Angular fixes their definition. Perhaps an option on the rule to allow decorators, or specific names?

**Comment by felpel** [comment]
Also note that this behavior occurs with the `Input` decorator as well. I didn't look at the definitions, but it should probably end with an `any` type at some point.

**Comment by felpel** [comment]
With @JoshuaKGoldberg's issue that was fixed on [Angular's side](https://github.com/angular/angular/issues/26942), I've tested in a new Angular application (v7) and their fix seems to work for the `Injectable` decorator since errors are no longer logged for injectable services.

If Angular's dependencies are up-to-date, then those `tslint`'s errors will be gone! 😄

TL;DR; `tslint` work as expected and found a bug in Angular's code.
